src/LaunchGrant.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { IGrant } from './IGrant.sol';
+
+contract LaunchGrant is IGrant {
+    uint256 internal immutable startWeeklyOffsetInSeconds   = 1682319600; // Monday, 24 April 2023 07:00:00
+    uint256 internal immutable launchDayTimestampInSeconds  = 1690182000; // Monday, 24 July 2023 07:00:00
+
+    function getCurrentId() external view override returns (uint256) {
+        uint currentTime = block.timestamp;
+        // Old grants have 1 week interval until Monday, 24 April 2023 06:59:59
+        if (currentTime < launchDayTimestampInSeconds) return (currentTime - startWeeklyOffsetInSeconds) / 1 weeks;
```

Comment:
Why do we need to keep expired grants around in the contract?

Reply:
I liked that the contract is also historic reference for grant amounts (during beta). But removed it now in favor of your other proposed implementation, since it's much cleaner :)